### PR TITLE
Add Adult Link in MM.

### PR DESCRIPTION
**Pull request by ZoeyZolotova**

### CODE REVIEW

File: packages/core/include/combo/common/api.h [hunk 1]
```diff
@@ -119,6 +119,8 @@ void DmaCompressed(u32 pstart, void* dst, u32 size);
 #if defined(GAME_OOT)
 void _DmaCompressed(u32 pstart, void* dst, u32 size);
 #endif
+s32 RequestDma(DmaRequest* request, void* vramStart, u32 vromStart, size_t size, u32 unused,
+                           OSMesgQueue* queue, OSMesg msg);
```
Comment (Nax) [inline packages/core/include/combo/common/api.h:123]: Style: single line

File: packages/core/include/combo/mm/object.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#ifndef INCLUDED_COMBO_MM_OBJECT_H
+#define INCLUDED_COMBO_MM_OBJECT_H
+
+#include <assert.h>
+#include <combo/util.h>
+
+#define OBJECT_EXCHANGE_BANK_MAX 35
+
+typedef struct {
```
Comment (Nax) [inline packages/core/include/combo/mm/object.h:9]: * Braces on their own lines
* No offset comments - use assertions instead
* No size comment, use a size assertion

File: packages/core/lib/combo/settings/data.ts [hunk 3]
```diff
@@ -881,6 +881,13 @@ export const SETTINGS = [{
   type: 'boolean',
   description: 'Makes the Mask Shop in Market open during the night',
   default: false
+}, {
+  key: 'crossAge',
+  name: 'Cross-Games Age',
+  category: 'main.cross',
+  type: 'boolean',
+  description: 'When you enter MM as Adult Link, you will be Adult Link in MM. Model credit: Skilar.',
```
Comment (Nax) [inline packages/core/lib/combo/settings/data.ts:889]: Credits shouldn't be here, or else people are gonna start adding credits for every asset/texture/logic/whatnot and it'll get unreadable (see old BSD software where the credits is larger than the documentation).

The model file should go in its own commit, commited by you but authored by Skilar, or mentionned in the commit descr, or both.

Once we add in-game credits we can list him there too.

File: packages/core/src/mm/scene.c [hunk 4]
```diff
@@ -20,3 +21,145 @@ PATCH_CALL(0x8013231c, Entrance_GetTableEntryCustom);
...
+            }
+        }
+
+        entry++;
+    }
+}
+
+PATCH_FUNC(0x8012F4FC, Object_UpdateEntriesCustom);
+
+void Object_AfterInitContext()
```
Comment (Nax) [inline packages/core/src/mm/scene.c:162]: Empty params list and `(void)` are not the same in C, this needs to be `(void)`

File: packages/core/src/mm/scene.c [hunk 5]
```diff
@@ -20,3 +21,145 @@ PATCH_CALL(0x8013231c, Entrance_GetTableEntryCustom);
...
+
+    addr = ((u32)objectCtx->slots[slot].segment) + vromSize;
+    addr = ALIGN16(addr);
+
+    return (void*)addr;
+}
+
+PATCH_FUNC(0x8012f73c, Object_AllocateSlotCustom);
+
+void Object_UpdateEntriesCustom(ObjectContext* objectCtx)
```
Comment (Nax) [inline packages/core/src/mm/scene.c:122]: Not sure I understand the point of this function, isn't it pretty much what the vanilla game is doing?

**Review by Nax — CHANGES_REQUESTED** [review]

File: packages/core/src/mm/scene.c (hunk 5 above)
Comment (ZoeyZolotova) [reply packages/core/src/mm/scene.c:122]: It's pretty much the same except that instead of `objectFile = &gObjectTable[id];` we use `objectFile = comboGetObjectData(id);` so that non-vanilla objects can work with this function.

File: packages/core/lib/combo/settings/data.ts (hunk 3 above)
Comment (ZoeyZolotova) [reply packages/core/lib/combo/settings/data.ts:889]: Ok I will amend the commits.
